### DISCUSSION TITLE
How to contribute in this case?

### DISCUSSION THREAD

**Discussion by YerePhy** [Lightning-AI/pytorch-lightning#12947] [General]
Hi, everyone.

**My background**
I come from physics, I have MSc in Data Science so I use Lightning as a user. I don't know how it works under the hood and how Lightning manage multi GPU set up or cluster training, this is really frustrating for me. Due to my bckg, I have no knowledge about  Computer Science realted stuff, I am not familiar with things like processes, threads, multithread, parallel computing, scheduling, ports, memory management, and a loooooong etc.

**So...**
Do you have any recommendations on how to start if I want to contribute to Lightning/Pytorch? Is it impossible/not feasible to learn all these things on the fly or how should I approach this huge gap?

Thanks in advance.

**Comment by pengzhangzhi** [comment]
As CS student I may share with you my personal experiences of learning CS. I think learning things like Concurrency (processes and threads) are important for writing large and time-consuming programs. Concurrency is effective to speed them up by a large margin. I am a self-taught student since my first year in university. I have collected a list of useful cs courses, which are free online. You can check out them in my homepage. Here is the link: https://pengzhangzhi.github.io/self-taught/
Among the courses I listed, I think the following courses may be helpful for you:

MIT6.828 Operating System Engineering
MIT 6.031: Software Construction

Both of them talk about concurrency in their lectures.
Hope it helps.

**Reply by YerePhy** [reply]
Okay @pengzhangzhi, I'm following [OSSU](https://github.com/ossu/computer-science) I hope one day I know a bit more about computers... the problem is that I like too many things...

**Reply by pengzhangzhi** [reply]
Great! Hope you have fun learning cs~
